src/solvers/dg/dg_2d.jl

```diff
@@ -363,61 +367,188 @@ function calc_volume_integral!(du::AbstractArray{<:Any,4}, u, nonconservative_te
+                                 inverse_weights[j] * (fstar2_L[v, i, j+1] - fstar2_R[v, i, j])) )
     end
   end
 
   return nothing
 end
 
-@inline function calcflux_fv!(fstar1, fstar2, u::AbstractArray{<:Any,4},
-                              equations, volume_flux_fv, dg::DGSEM, element)
 
-  fstar1[:, 1,            :] .= zero(eltype(fstar1))
-  fstar1[:, nnodes(dg)+1, :] .= zero(eltype(fstar1))
+"""
+    calcflux_fv!(fstar1_L, fstar1_R, fstar2_L, fstar2_R, u_leftright,
+                 nonconservative_terms::Val{false}, equations, volume_flux_fv, dg, element_id)
+
+Calculate the finite volume fluxes inside the elements (without non-conservative terms).
+
+# Arguments
+- `fstar1_L::AbstractArray{T} where T<:Real`:
+- `fstar1_R::AbstractArray{T} where T<:Real`:
+- `fstar2_L::AbstractArray{T} where T<:Real`:
+- `fstar2_R::AbstractArray{T} where T<:Real`:
+- `u_leftright::AbstractArray{T} where T<:Real`
+- `nonconservative_terms::Logical`
+- `equations`
+- `volume_flux_fv`
+- `dg::DGSEM`
+- `element_id::Integer`
+"""
+@inline function calcflux_fv!(fstar1_L, fstar1_R, fstar2_L, fstar2_R, u::AbstractArray{<:Any,4},
+                              nonconservative_terms::Val{false}, equations, volume_flux_fv,
+                              dg::DGSEM, element)
+
+  fstar1_L[:, 1,            :] .= zero(eltype(fstar1_L))
+  fstar1_L[:, nnodes(dg)+1, :] .= zero(eltype(fstar1_L))
+  fstar1_R[:, 1,            :] .= zero(eltype(fstar1_R))
+  fstar1_R[:, nnodes(dg)+1, :] .= zero(eltype(fstar1_R))
 
   for j in eachnode(dg), i in 2:nnodes(dg)
     u_ll = get_node_vars(u, equations, dg, i-1, j, element)
     u_rr = get_node_vars(u, equations, dg, i,   j, element)
     flux = volume_flux_fv(u_ll, u_rr, 1, equations) # orientation 1: x direction
-    set_node_vars!(fstar1, flux, equations, dg, i, j)
+    set_node_vars!(fstar1_L, flux, equations, dg, i, j)
+    set_node_vars!(fstar1_R, flux, equations, dg, i, j)
   end
 
-  fstar2[:, :, 1           ] .= zero(eltype(fstar2))
-  fstar2[:, :, nnodes(dg)+1] .= zero(eltype(fstar2))
+  fstar2_L[:, :, 1           ] .= zero(eltype(fstar2_L))
+  fstar2_L[:, :, nnodes(dg)+1] .= zero(eltype(fstar2_L))
+  fstar2_R[:, :, 1           ] .= zero(eltype(fstar2_R))
+  fstar2_R[:, :, nnodes(dg)+1] .= zero(eltype(fstar2_R))
 
   for j in 2:nnodes(dg), i in eachnode(dg)
     u_ll = get_node_vars(u, equations, dg, i, j-1, element)
     u_rr = get_node_vars(u, equations, dg, i, j,   element)
     flux = volume_flux_fv(u_ll, u_rr, 2, equations) # orientation 2: y direction
-    set_node_vars!(fstar2, flux, equations, dg, i, j)
+    set_node_vars!(fstar2_L, flux, equations, dg, i, j)
+    set_node_vars!(fstar2_R, flux, equations, dg, i, j)
   end
 
   return nothing
 end
 
+"""
+    calcflux_fv!(fstar1_L, fstar1_R, fstar2_L, fstar2_R, u_leftright,
+                 nonconservative_terms::Val{true}, equations, volume_flux_fv, dg, element_id)
+
+Calculate the finite volume fluxes inside the elements (non-conservative terms are active).
```

Comment:
```suggestion
Calculate the finite volume fluxes inside the elements (**with non-conservative terms**).
```
Same reason as above